src/test.ts

```diff
@@ -104,9 +109,36 @@ export function runTest(testDirectory: string, rulesDirectory?: string | string[
             };
         });
 
+        // test against fixed files
+        let fixedFileText: string;
+        let newFileText: string;
+        try {
+            const fixedFile = fileToLint.replace(/\.lint$/, FIXES_FILE_EXTENSION);
+            const stat = fs.statSync(fixedFile);
+            if (stat.isFile()) {
+                fixedFileText = fs.readFileSync(fixedFile, "utf8");
+                // accumulate replacements
+                let replacements: Replacement[] = [];
+                for (const failure of failures) {
+                    const fixes = failure.getFixes();
+                    if (fixes.length > 0) {
+                        replacements = replacements.concat(fixes[0].getReplacements());
+                    }
+                }
+                // sort in reverse so that diffs are properly applied
+                replacements.sort((a, b) => b.getEnd() - a.getEnd());
```

Comment:
This logic still seems somewhat duplicated to me, maybe there should be a static `Fix.applyAll(text: string, fixes: Fix[])` method?